### PR TITLE
Update vertica

### LINKED ISSUE #11022: Update to newer Vertica DB for tests
Right now we're testing against Vertica 7.1.2, but 9.2.1 is out. 

We'll need to use a newer Docker image for CI. The one we're currently using seems like it's no longer being actively developed/ [jbfavre/vertica:latest](https://github.com/jbfavre/docker-vertica) looks like a reasonable replacement, I was able to drop it in without having to change any env vars. We will also need to switch the JDBC driver version used in CI.

I switched to a newer 9.x JDBC driver, but I've ran into a few problems when running tests: 

```
Caused by: com.vertica.support.exceptions.DataException: [Vertica][VJDBC](3787) ROLLBACK: Invalid value for parameter
 at com.vertica.util.ServerErrorData.buildException (:-1)
    com.vertica.dataengine.VResultSet.fetchChunk (:-1)
    com.vertica.dataengine.VResultSet.initialize (:-1)
    com.vertica.dataengine.VQueryExecutor.readExecuteResponse (:-1)
    com.vertica.dataengine.VQueryExecutor.handleExecuteResponse (:-1)
    com.vertica.dataengine.VQueryExecutor.execute (:-1)
    com.vertica.jdbc.common.SPreparedStatement.executeWithParams (:-1)
    com.vertica.jdbc.common.SPreparedStatement.executeUpdate (:-1)
```

This was happening when loading the test data. So we'll have to work out why that's happening in order to upgrade.

### PR DESCRIPTION
**Pull request by metamben**
Fixes #11022.

Use the `vertica/vertica-ce:12.0.2-0` image.

### CODE REVIEW

File: .github/workflows/drivers.yml [hunk 1]
```diff
@@ -663,15 +663,13 @@ jobs:
...
-        image: sumitchawla/vertica
+        image: vertica/vertica-ce:12.0.2-0
         ports:
           - "5433:5433"
     steps:
     - uses: actions/checkout@v3
     - name: Make plugins directory
       run: mkdir plugins
-    - name: Fetch JDBC driver
-      run: wget --output-document=plugins/vertica-jdbc-7.1.2-0.jar ${{ secrets.VERTICA_JDBC_JAR }}
```
Comment (metamben) [inline .github/workflows/drivers.yml:674]: I'm not quite sure why this was needed, the vertica plugin had a dependency on `com.vertica.jdbc/vertica-jdbc {:mvn/version "10.0.1-0"}`.

File: test/metabase/query_processor_test/share_test.clj [hunk 2]
```diff
@@ -32,14 +32,22 @@
...
-      ;; due to a bug in the Mongo counts are returned as empty when there are no results (#5419)
-      (is (= (if (= driver/*driver* :mongo)
-               []
-               [[nil]])
-             (mt/rows
-               (mt/run-mbql-query venues
-                 {:aggregation [[:share [:< $price 4]]]
-                  :filter      [:> $price Long/MAX_VALUE]})))))))
+      ;; Vertica doesn't allow dividing null by zero
+      ;; TODO consider wrapping all divisions in nullif checking the first argument
```
Comment (metamben) [inline test/metabase/query_processor_test/share_test.clj:36]: Is it worth wrapping every division to be compatible with other DBs?

File: .github/workflows/drivers.yml (hunk 1 above)
Comment (camsaul) [reply .github/workflows/drivers.yml:674]: At some point the Vertica driver was not available on Maven, I guess we forgot to remove this code once we were able to add the dependency

File: test/metabase/query_processor_test/share_test.clj (hunk 2 above)
Comment (camsaul) [reply test/metabase/query_processor_test/share_test.clj:36]: I don't think most other databases support dividing by zero. So if we're running into divide by zero errors it's probably a symptom of something else. `[:> $price Long/MAX_VALUE]` should mean that no rows match, so doing a `:share` aggregation over no rows shouldn't even be trying to divide anything

Comment (metamben) [reply test/metabase/query_processor_test/share_test.clj:36]: Well, doesn't the fact that the other databases in `(disj (mt/normal-drivers-with-feature :basic-aggregations) :mongo)` pass this test imply that most DBs support `null / 0` (and return `null`)?

**Review by camsaul — APPROVED** [review]

**Pull request merged**